js/store.js

```diff
@@ -0,0 +1,77 @@
+const store = {
+  init() {
+    this.addListeners();
+  },
+
+  getAll() {
+    return this.get();
+  },
+
+  getFirstParties() {
+    /*
+    * similar to getAll() without the thirPartyRequests
+    */
+  },
+
+  async getFirstParty(hostname) {
+    if (!hostname) {
+      return null;
+    }
+
+    const storage = await this.get();
+    return storage.websites[`${hostname}`];
+  },
+
+  async getThirdParties(hostname) {
+    if (!hostname) {
+      return null;
+    }
+
+    const storage = await this.get();
+
+    return storage.websites[`${hostname}`] ?
+      storage.websites[`${hostname}`].thirdPartyRequests :
+      null;
+  },
+
+  setFirstParty(param) {
+    /**
+     * @param object with hostname as key
+     * @todo validate the param
+     */
+    return this.set(param);
+  },
+
+  async get(params) {
```

Comment:
This code probably belongs better in `getAll()` we can add this later if we need it.